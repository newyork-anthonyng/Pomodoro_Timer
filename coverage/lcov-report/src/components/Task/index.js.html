<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/Task/index.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">src/components/Task</a> index.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>8/8</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>8/8</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">21x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import T from 'prop-types';
import noop from '../../utility/noop';
import glamorous from 'glamorous';
&nbsp;
const H4 = glamorous.h4({
  display: 'inline-block',
  margin: 0,
});
&nbsp;
const P = glamorous.p({
  marginTop: '10px',
});
&nbsp;
const Button = glamorous.button({
  float: 'right',
});
&nbsp;
const Task = ({ id, title, text, onDeleteClick }) =&gt; (
  &lt;div data-taskId={id}&gt;
    &lt;H4&gt;{title}&lt;/H4&gt;
&nbsp;
    &lt;Button onClick={() =&gt; onDeleteClick(id)}&gt;Ã—&lt;/Button&gt;
&nbsp;
    &lt;P&gt;{text}&lt;/P&gt;
  &lt;/div&gt;
);
&nbsp;
Task.defaultProps = {
  onDeleteClick: noop,
};
&nbsp;
Task.propTypes = {
  id: T.string.isRequired,
  title: T.string.isRequired,
  text: T.string.isRequired,
  onDeleteClick: T.func.isRequired,
};
&nbsp;
export default Task;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sat May 06 2017 15:20:25 GMT-0700 (PDT)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
