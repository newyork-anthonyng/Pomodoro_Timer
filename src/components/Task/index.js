import React from 'react';
import T from 'prop-types';
import noop from '../../utility/noop';
import glamorous from 'glamorous';

const H4 = glamorous.h4({
  display: 'inline-block',
  margin: 0,
});

const P = glamorous.p({
  marginTop: '10px',
});

const Button = glamorous.button({
  float: 'right',
});

const Task = ({ id, title, text, onDeleteClick }) => (
  <div data-taskId={id}>
    <H4>{title}</H4>

    <Button onClick={() => onDeleteClick(id)}>Ã—</Button>

    <P>{text}</P>
  </div>
);

Task.defaultProps = {
  onDeleteClick: noop,
};

Task.propTypes = {
  id: T.string.isRequired,
  title: T.string.isRequired,
  text: T.string.isRequired,
  onDeleteClick: T.func.isRequired,
};

export default Task;
