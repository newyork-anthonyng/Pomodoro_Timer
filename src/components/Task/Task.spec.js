import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Task from './';
import noop from '../../utility/noop';

const defaultProps = {
  id: 'superSecretId',
  title: 'Task 1',
  text: 'Watch Netflix',
  onDeleteClick: noop,
};

it('should render', () => {
  const wrapper = shallow(
    <Task {...defaultProps} />
  );
  expect(toJson(wrapper)).toMatchSnapshot();
});

it('should run callback when clicking on delete button', () => {
  const cb = jest.fn();
  const wrapper = mount(
    <Task
      {...defaultProps}
      onDeleteClick={cb}
    />
  );
  const button = wrapper.find('button');
  button.simulate('click');

  expect(cb.mock.calls.length).toEqual(1);
  expect(cb.mock.calls[0][0]).toEqual(defaultProps.id);
});
