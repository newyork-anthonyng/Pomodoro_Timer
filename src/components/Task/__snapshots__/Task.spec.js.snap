// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div
  data-taskId="superSecretId"
>
  <glamorous(h4)>
    Task 1
  </glamorous(h4)>
  <glamorous(button)
    onClick={[Function]}
  >
    Ã—
  </glamorous(button)>
  <glamorous(p)>
    Watch Netflix
  </glamorous(p)>
</div>
`;
